### PR TITLE
DEVDOCS-4891: [new] Catalog API, add handling for inventory limits

### PR DESCRIPTION
**Pull request by bc-andreadao**
# [DEVDOCS-4891] 

**_This PR must be merged AFTER [PR 161](https://github.com/bigcommerce/api-specs-ssot/pull/161) to prevent broken heading links_**

## What changed?
- Add how inventory for CATALOG API is handled when inventory exceeds limit

## Anything else?
Related to [PR 161](https://github.com/bigcommerce/api-specs-ssot/pull/161)

[DEVDOCS-4891]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4891?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reference/catalog/product-variants_catalog.v3.yml [hunk 1]
```diff
@@ -1272,7 +1272,12 @@ paths:
                               x-nullable: true
                             inventory_level:
                               type: integer
-                              description: 'Inventory level for the variant, which is used when the product’s inventory_tracking is set to `variant`.'
+                              description: |- 
+                                Current inventory level of the product. Simple inventory tracking must be enabled (See the `inventory_tracking` field) for this to take any effect.
```
Comment (alex-gaiduchok) [inline reference/catalog/product-variants_catalog.v3.yml:1276]: actually you don’t need tracking mode to be enabled. UI hides form but API endpoints(both catalog and inventory) allows to set stock for untracked products

Comment (bc-andreadao) [reply reference/catalog/product-variants_catalog.v3.yml]: @alex-gaiduchok Thanks, I accidentally changed the sentence, so I just changed it back to how it was before

**Review by alex-gaiduchok — APPROVED** [review]

File: reference/catalog/product-variants_catalog.v3.yml [hunk 2]
```diff
@@ -1272,7 +1272,14 @@ paths:
...
-                              description: 'Inventory level for the variant, which is used when the product’s inventory_tracking is set to `variant`.'
+                              description: |- 
+                                Inventory level for the variant, which is used when the product’s inventory_tracking is set to `variant`. The Catalog API returns the inventory for only the default location.
+
+                                The inventory for a variant cannot exceed 2,147,483,647 in the catalog. The sum of the variant inventories, or the total inventory for a product, cannot exceed 2,147,483,647. 
+                                
+                                If you exceed the limit, the store sets the variant inventory to the limit if no other variant inventories are set. If other variant inventories are set, the store does not save the variant inventory rather than setting the variant inventory to the remaining limit.
+
+                                The Catalog API handles limits in a different way than the Inventory API. For more, see [Limit handling](/buy-online-pick-up-in-store/inventory#limit-handling-in-inventory-versus-catalog-api).
```
Comment (bc-traciporter) [inline reference/catalog/product-variants_catalog.v3.yml:1282]:
```suggestion
                                The Catalog API handles limits in a different way than the Inventory API. For more information, see [Limit handling](/buy-online-pick-up-in-store/inventory#limit-handling-in-inventory-versus-catalog-api). 
```

File: reference/catalog/product-variants_catalog.v3.yml [hunk 3]
```diff
@@ -1460,7 +1467,14 @@ paths:
...
                         type: integer
-                        description: 'Inventory level for the variant, which is used when the product’s inventory_tracking is set to `variant`.'
+                        description: |- 
+                          Inventory level for the variant, which is used when the product’s inventory_tracking is set to `variant`. The Catalog API returns the inventory for only the default location.
+
+                          The inventory for a variant cannot exceed 2,147,483,647 in the catalog. The sum of the variant inventories, or the total inventory for a product, cannot exceed 2,147,483,647. 
+                          
+                          If you exceed the limit, the store sets the variant inventory to the limit if no other variant inventories are set. If other variant inventories are set, the store does not save the variant inventory rather than setting the variant inventory to the remaining limit.
+
+                          The Catalog API handles limits in a different way than the Inventory API. For more, see [Limit handling](/buy-online-pick-up-in-store/inventory#limit-handling-in-inventory-versus-catalog-api).
```
Comment (bc-traciporter) [inline reference/catalog/product-variants_catalog.v3.yml:1477]:
```suggestion
                          The Catalog API handles limits in a different way than the Inventory API. For more information, see [Limit handling](/buy-online-pick-up-in-store/inventory#limit-handling-in-inventory-versus-catalog-api). 
```

File: reference/catalog/product-variants_catalog.v3.yml [hunk 4]
```diff
@@ -1575,7 +1589,14 @@ paths:
...
-                              description: 'Inventory level for the variant, which is used when the product’s inventory_tracking is set to `variant`.'
+                              description: |- 
+                                Inventory level for the variant, which is used when the product’s inventory_tracking is set to `variant`. The Catalog API returns the inventory for only the default location.
+
+                                The inventory for a variant cannot exceed 2,147,483,647 in the catalog. The sum of the variant inventories, or the total inventory for a product, cannot exceed 2,147,483,647. 
+                                
+                                If you exceed the limit, the store sets the variant inventory to the limit if no other variant inventories are set. If other variant inventories are set, the store does not save the variant inventory rather than setting the variant inventory to the remaining limit.
+
+                                The Catalog API handles limits in a different way than the Inventory API. For more, see [Limit handling](/buy-online-pick-up-in-store/inventory#limit-handling-in-inventory-versus-catalog-api).
```
Comment (bc-traciporter) [inline reference/catalog/product-variants_catalog.v3.yml:1599]:
```suggestion
                                The Catalog API handles limits in a different way than the Inventory API. For more information, see [Limit handling](/buy-online-pick-up-in-store/inventory#limit-handling-in-inventory-versus-catalog-api). 
```

File: reference/catalog/product-variants_catalog.v3.yml [hunk 5]
```diff
@@ -2061,7 +2089,14 @@ components:
...
               type: integer
-              description: 'Inventory level for the variant, which is used when the product’s inventory_tracking is set to `variant`.'
+              description: |- 
+                Inventory level for the variant, which is used when the product’s inventory_tracking is set to `variant`. The Catalog API returns the inventory for only the default location.
+
+                The inventory for a variant cannot exceed 2,147,483,647 in the catalog. The sum of the variant inventories, or the total inventory for a product, cannot exceed 2,147,483,647. 
+                
+                If you exceed the limit, the store sets the variant inventory to the limit if no other variant inventories are set. If other variant inventories are set, the store does not save the variant inventory rather than setting the variant inventory to the remaining limit.
+
+                The Catalog API handles limits in a different way than the Inventory API. For more, see [Limit handling](/buy-online-pick-up-in-store/inventory#limit-handling-in-inventory-versus-catalog-api).
```
Comment (bc-traciporter) [inline reference/catalog/product-variants_catalog.v3.yml:2099]:
```suggestion
                The Catalog API handles limits in a different way than the Inventory API. For more information, see [Limit handling](/buy-online-pick-up-in-store/inventory#limit-handling-in-inventory-versus-catalog-api). 
```

File: reference/catalog/product-variants_catalog.v3.yml [hunk 6]
```diff
@@ -1874,7 +1895,14 @@ components:
...
           type: integer
-          description: 'Inventory level for the variant, which is used when the product’s inventory_tracking is set to `variant`.'
+          description: |- 
+            Inventory level for the variant, which is used when the product’s inventory_tracking is set to `variant`. The Catalog API returns the inventory for only the default location.
+
+            The inventory for a variant cannot exceed 2,147,483,647 in the catalog. The sum of the variant inventories, or the total inventory for a product, cannot exceed 2,147,483,647. 
+            
+            If you exceed the limit, the store sets the variant inventory to the limit if no other variant inventories are set. If other variant inventories are set, the store does not save the variant inventory rather than setting the variant inventory to the remaining limit.
+
+            The Catalog API handles limits in a different way than the Inventory API. For more, see [Limit handling](/buy-online-pick-up-in-store/inventory#limit-handling-in-inventory-versus-catalog-api).
```
Comment (bc-traciporter) [inline reference/catalog/product-variants_catalog.v3.yml:1905]:
```suggestion
            The Catalog API handles limits in a different way than the Inventory API. For more information, see [Limit handling](/buy-online-pick-up-in-store/inventory#limit-handling-in-inventory-versus-catalog-api). 
```

File: reference/catalog/products_catalog.v3.yml [hunk 7]
```diff
@@ -6659,7 +6659,14 @@ components:
...
           type: integer
-          description: 'Inventory level for the variant, which is used when the product’s inventory_tracking is set to `variant`.'
+          description: |- 
+            Inventory level for the variant, which is used when the product’s inventory_tracking is set to `variant`. The Catalog API returns the inventory for only the default location.
+                                
+            The inventory for a variant cannot exceed 2,147,483,647 in the catalog. The sum of the variant inventories, or the total inventory for a product, cannot exceed 2,147,483,647. 
+                                
+            If you exceed the limit, the store sets the variant inventory to the limit if no other variant inventories are set. If other variant inventories are set, the store does not save the variant inventory rather than setting the variant inventory to the remaining limit.
+            
+            The Catalog API handles limits in a different way than the Inventory API. For more, see [Limit handling](/buy-online-pick-up-in-store/inventory#limit-handling-in-inventory-versus-catalog-api).
```
Comment (bc-traciporter) [inline reference/catalog/products_catalog.v3.yml:6669]:
```suggestion
            The Catalog API handles limits in a different way than the Inventory API. For more information, see [Limit handling](/buy-online-pick-up-in-store/inventory#limit-handling-in-inventory-versus-catalog-api). 
```

File: reference/catalog/products_catalog.v3.yml [hunk 8]
```diff
@@ -6842,7 +6849,14 @@ components:
...
```
Comment (bc-traciporter) [inline reference/catalog/products_catalog.v3.yml:6859]:
```suggestion
            The Catalog API handles limits in a different way than the Inventory API. For more information, see [Limit handling](/buy-online-pick-up-in-store/inventory#limit-handling-in-inventory-versus-catalog-api). 
```

File: reference/catalog/products_catalog.v3.yml [hunk 9]
```diff
@@ -8271,8 +8285,12 @@ components:
...
           minimum: 0
           type: integer
-          description: |
-            Current inventory level of the product. Simple inventory tracking must be enabled (See the `inventory_tracking` field) for this to take any effect.
+          description: |- 
+            Current inventory level of the product. You must track inventory by _product_ for this to take effect (see the `inventory_tracking` field). The Catalog API returns the inventory for only the default location.
+
+            The inventory for a product cannot exceed 2,147,483,647 in the catalog. If you exceed the limit, the store sets the inventory level to the limit.
+
+            The Catalog API handles limits in a different way than the Inventory API. For more, see [Limit handling](/buy-online-pick-up-in-store/inventory#limit-handling-in-inventory-versus-catalog-api).
```
Comment (bc-traciporter) [inline reference/catalog/products_catalog.v3.yml:8293]:
```suggestion
            The Catalog API handles limits in a different way than the Inventory API. For more information, see [Limit handling](/buy-online-pick-up-in-store/inventory#limit-handling-in-inventory-versus-catalog-api). 
```

**Review by bc-traciporter — DISMISSED** [review]
Looks good, I added one editorial comment.

**Review by bc-traciporter — APPROVED** [review]

**Pull request merged**